File: src/src_core/fmg/core/types/ClickCellResult.java
package fmg.core.types;

import java.util.HashSet;
import java.util.Set;

import fmg.core.mosaic.cells.BaseCell;

public class ClickCellResult {

   public Set<BaseCell> modified = new HashSet<BaseCell>();

   /** были ли изменения на поле? Была ли открыта хоть одна ячейка или выставлен/снят флажёк / знак вопроса */
   public boolean isAnyChanges() { return !modified.isEmpty(); }
   public boolean isAnyOpenMine() {
      return modified.stream()
            .filter(x -> x.getState().getStatus() == EState._Open)
            .anyMatch(x -> x.getState().getOpen() == EOpen._Mine);
   }

//   /** множество ячеек (нулевых  ) открытых при последнем клике */
//   public Set<BaseCell> getOpenNils() {
//      return modified.stream()
//            .filter(x -> x.getState().getStatus() == EState._Open)
//            .filter(x -> x.getState().getOpen() == EOpen._Nil)
//            .collect(Collectors.toSet());
//   }
//
//   /** множество ячеек (ненулевых) открытых при последнем клике */
//   public Set<BaseCell> getOpens() {
//      return modified.stream()
//            .filter(x -> x.getState().getStatus() == EState._Open)
//            .filter(x -> x.getState().getOpen() != EOpen._Nil)
//            .collect(Collectors.toSet());
//   }
//
//   /** множество ячеек с флажками снятых/уставленных при последнем клике */
//   public Set<BaseCell> getFlags() {
//      return modified.stream()
//            .filter(x -> x.getState().getStatus() == EState._Close)
//            .collect(Collectors.toSet());
//   }

   public int getCountFlag() {
      return (int)modified.stream()
            .filter(x -> x.getState().getStatus() == EState._Close)
            .filter(x -> x.getState().getClose() == EClose._Flag)
            .count();
   }

   public int getCountOpen() {
      return (int)modified.stream()
            .filter(x -> x.getState().getStatus() == EState._Open)
            .count();
   }

   public int getCountUnknown() {
      return (int)modified.stream()
            .filter(x -> x.getState().getStatus() == EState._Close)
            .filter(x -> x.getState().getClose() == EClose._Unknown)
            .count();
   }

}

File: src/src_core/fmg/core/types/ClickCellResult.cs
using System.Collections.Generic;
using System.Linq;
using fmg.core.mosaic.cells;

namespace fmg.core.types.click {

   public class ClickCellResult {

      public ISet<BaseCell> Modified = new HashSet<BaseCell>();

      /// <summary> были ли изменения на поле? Была ли открыта хоть одна ячейка или выставлен/снят флажёк / знак вопроса </summary>
      public bool IsAnyChanges => Modified.Any();
      public bool IsAnyOpenMine => Modified
         .Where(x => x.State.Status == EState._Open)
         .Any(x => x.State.Open == EOpen._Mine);

      ///// <summary> множество ячеек (нулевых) открытых при последнем клике </summary>
      //public List<BaseCell> GetOpenNils() {
      //   return Modified
      //      .Where(x => x.State.Status == EState._Open)
      //      .Where(x => x.State.Open == EOpen._Nil)
      //      .ToList();
      //}

      ///// <summary> множество ячеек (ненулевых) открытых при последнем клике </summary>
      //public List<BaseCell> GetOpens() {
      //   return Modified
      //      .Where(x => x.State.Status == EState._Open)
      //      .Where(x => x.State.Open != EOpen._Nil)
      //      .ToList();
      //}

      ///// <summary> множество ячеек с флажками снятых/уставленных при последнем клике </summary>
      //public List<BaseCell> GetFlags() {
      //   return Modified
      //      .Where(x => x.State.Status == EState._Close)
      //      .ToList();
      //}

      public int CountFlag => Modified
         .Where(x => x.State.Status == EState._Close)
         .Count(x => x.State.Close == EClose._Flag);

      public int CountOpen => Modified
         .Count(x => x.State.Status == EState._Open);

      public int CountUnknown => Modified
         .Where(x => x.State.Status == EState._Close)
         .Count(x => x.State.Close == EClose._Unknown);

   }

}
